client/src/components/InputsModal/InputsModal.tsx

```diff
@@ -133,22 +136,131 @@ const InputsModal: FC<InputsModalProps> = ({
     }
   });
 
+  const [inputType, setInputType] = React.useState<string>('Field');
+  const [baseInput, setBaseInput] = React.useState<string>('');
+  const [invalidInput, setInvalidInput] = React.useState<boolean>(false);
+
+  useEffect(() => {
+    const serialObj = inputs.map((requirement: TestInput) => {
+      if (requirement.type == 'oauth_credentials') {
+        return {
+          ...requirement,
+          value:
+            (JSON.parse(inputsMap.get(requirement.name) as string) as OAuthCredentials) || '{}',
+        };
+      } else if (requirement.type == 'radio') {
+        const firstVal =
+          requirement.options?.list_options && requirement.options?.list_options?.length > 0
+            ? requirement.options?.list_options[0]?.value
```

Comment:
```suggestion
            ? requirement.options.list_options[0]?.value
```

Nit. The conditional above already confirms that options.list_options is not nullish and contains at least one element in the array